src/traversals/bipartiteness.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    bipartite_map(g)
+
+For a bipartite graph `g`, return a vector `c` of size ``|V|`` containing
+the assignment of each vertex to one of the two sets (``c_i == 1`` or c_i == 2``).
+If `g` is not bipartite, return an empty vector.
+
+### Implementation Notes
+Note that an empty vector does not necessarily indicate non-bipartiteness.
+An empty graph will return an empty vector but is bipartite.
+"""
+function bipartite_map(g::AbstractGraph)
+    T = eltype(g)
+    nvg = nv(g)
+    if !is_directed(g)
+        ccs = filter(x -> length(x) > 2, connected_components(g))
+    else
+        ccs = filter(x -> length(x) > 2, weakly_connected_components(g))
+    end
+    seen = falses(nvg)
+    colors = falses(nvg)
```

Comment:
Actually, I'm not sure I want to change this, because L31 is a nice way of handling things.